pkg/controller/atlasfederatedauth/atlasfederatedauth.go

```diff
@@ -0,0 +1,124 @@
+package atlasfederatedauth
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"go.mongodb.org/atlas/mongodbatlas"
+
+	"github.com/google/go-cmp/cmp"
+
+	mdbv1 "github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/controller/workflow"
+)
+
+const (
+	ErrInvalidProvider    = "INVALID_PROVIDER"
+	ErrNotOrgGroupCreator = "NOT_ORG_GROUP_CREATOR"
```

Comment:
Did we have a place for all Atlas error codes?

Reply:
In `pkg/controller/atlas/api_error.go`